src/frontend/src/optimizer/property/distribution.rs

```diff
@@ -328,20 +347,27 @@ impl RequiredDist {
     }
 
     fn enforce(&self, plan: PlanRef, required_order: &Order) -> PlanRef {
+        let dist = self.to_dist(&plan);
+        match plan.convention() {
+            Convention::Batch => BatchExchange::new(plan, required_order.clone(), dist).into(),
+            Convention::Stream => StreamExchange::new(plan, dist).into(),
+            _ => unreachable!(),
+        }
+    }
+
+    fn to_dist(&self, plan: &impl GenericPlanRef) -> Distribution {
         let dist = match self {
+            // all the distribution satisfy the Any, and the function can be only called by
+            // `enforce_if_not_satisfies`
             RequiredDist::Any => unreachable!(),
             // TODO: add round robin distributed type
-            RequiredDist::AnyShard => todo!(),
+            RequiredDist::AnyShard => Distribution::HashShard(plan.logical_pk().to_vec()),
```

Comment:
prefer to leave a todo!(), for actually the brach is not achieved now. And if the logical_pk() is empty, maybe it is wrong.

Reply:
ura!

Reply:
fixed